cmd/kubelet/app/options/globalflags.go

```diff
@@ -37,11 +39,12 @@ import (
 // AddGlobalFlags explicitly registers flags that libraries (glog, verflag, etc.) register
 // against the global flagsets from "flag" and "github.com/spf13/pflag".
 // We do this in order to prevent unwanted flags from leaking into the Kubelet's flagset.
-func AddGlobalFlags(fs *pflag.FlagSet) {
-	addGlogFlags(fs)
-	addCadvisorFlags(fs)
-	verflag.AddFlags(fs)
-	logs.AddFlags(fs)
+// If fake is true, will register the flag name, but point it at a value with a noop Set implementation.
+func AddGlobalFlags(fs *pflag.FlagSet, fake bool) {
```

Comment:
propagating `fake` to all the flag registering functions seems really odd. If you want fake flags just to enable reparsing, can't you call AddGlobalFlags on an empty flag set `f1`, iterate over the flags that get added, and create correspondingly named fake flags in a flag set `f2` up in main() without changing anything else?

Reply:
Hmm, yes that would also work.

Reply:
Updated to do that instead.